### PR TITLE
Implement container-interop

### PR DESCRIPTION
**Pull request by gianarb**
This proposal is a simple implementation of ContainerInterface.
Now we can use it in [zend-stratigility-skeleton](https://github.com/container-interop/container-interop) for example.

### CODE REVIEW

**Comment by weierophinney** [comment]
My only concern is that the container-interop interfaces were just admitted as a draft PSR... which means there's still a very high liklihood of changes possible. We may want to do this as an "InteropManager" or similar, with documentation in the README indicating it's experimental and unstable until the PSR for container interop is accepted.

**Comment by gianarb** [comment]
I can try to write a `InteropManager` that extends `Zend\ServiceManager\ServiceManager` but extends `Interop\Container\ContainerInterface`

**Comment by gianarb** [comment]
@weierophinney this is a project's draft :) 
https://github.com/gianarb/zend-servicemanager-container-interop

easy easy

**Comment by weierophinney** [comment]
@gianarb That's exactly what I was suggesting!

Pinging @mvriel : is there a phpdoc annotation for indicating a class or method is "unstable"?

**Comment by gianarb** [comment]
In your opinion is a good idea builds a new project? 

:+1: It's clear in my opinion and we can not add unstable class in zend-servicemanager

**Comment by mvriel** [comment]
@weierophinney no, the best you could do is use to mark it using `@access protected` or `@access private`; though that tag is deprecated and removed from PSR-5

**Comment by sandrokeil** [comment]
Looks good. Maybe one problem could be the [optional parameters](https://github.com/zendframework/zend-servicemanager/blob/master/src/ServiceManager.php#L492 "ZF2 ServicerManager"). I know it's not a problem in PHP but it's not so clean. Isn't it?

**Comment by gianarb** [comment]
@sandrokeil the intereface is already implemented without modified.. Why could it be a problem?

**Comment by sandrokeil** [comment]
@gianarb I think it's confusing because the ContainerInterface function definition is `get($id)` and in ServiceManager we have `get($name, $usePeeringServiceManagers = true)`. Also in ZF3 it's planned to have [public function get($name, array $options = [])](https://github.com/bakura10/zend-servicemanager/blob/refactor/src/ServiceManager.php#L94).

The ServiceLocatorInterface is ok, that's right. But it could be changed in [ZF3](https://github.com/bakura10/zend-servicemanager/blob/refactor/src/ServiceLocatorInterface.php#L24). So it's magic to call `$sl->get('service', $options)` or `$sl->get('service', false)`.

So I think it's not real support of interoperability. But we could fix this for ZF3.

**Comment by Ocramius** [comment]
> My only concern is that the container-interop interfaces were just admitted as a draft PSR... 

`container-interop` as-is is a project on its own. It will maybe become a PSR, yes /cc @moufmouf

 > @gianarb I think it's confusing because the `ContainerInterface` function definition is `get($id)` and in ServiceManager we have `get($name, $usePeeringServiceManagers = true)`.

Note that this is perfectly legit PHP

**Comment by weierophinney** [comment]
Based on the feedback from @Ocramius, I'm :+1: - however, it looks like we need to add a little more test coverage, @gianarb - this small change has decreased coverage. :)

**Comment by weierophinney** [comment]
Merged to develop for release with 2.6.0.

File: composer.json [hunk 1]
```diff
@@ -13,7 +13,8 @@
         }
     },
     "require": {
-        "php": ">=5.5"
+        "php": ">=5.5",
+        "container-interop/container-interop": "*"
```
Comment (Ocramius) [inline composer.json]: Note: this should be a constrained version (1.0.*)

Comment (gianarb) [inline composer.json]: Sorry #5

**Pull request closed without merging**